CMakeLists.txt

```
@@ -1,248 +1,74 @@
-if(USE_SYSTEM_PYMAVLINK)
-    find_program (
-        mavgen_EXECUTABLE
-        NAMES mavgen mavgen.py
-    )
-
-    # Fallback to Python invocation, if not found.
-    if (mavgen_EXECUTABLE-NOTFOUND)
-        set(mavgen_EXECUTABLE ${mavgen_EXECUTABLE_WITH_PYTHON_INVOCATION})
-    endif()
-else()
-    set(mavgen_EXECUTABLE ${mavgen_EXECUTABLE_WITH_PYTHON_INVOCATION})
-endif()
-message(STATUS "mavgen_EXECUTABLE: ${mavgen_EXECUTABLE}")
-
-macro(generateMavlink version definitions)
-    foreach(definition ${definitions})
-        set(targetName ${definition}-v${version})
-        set(definitionAbsPath ${MAVLINK_SOURCE_DIR}/message_definitions/v1.0/${definition})
-        message(STATUS "processing: ${definitionAbsPath}")
-        add_custom_command( 
-            OUTPUT ${targetName}-stamp
-            COMMAND ${mavgen_EXECUTABLE} --lang=C
-                                         --wire-protocol=${version}
-                                         --output=include/v${version}
-                                         ${definitionAbsPath}
-            COMMAND touch ${targetName}-stamp
-        )
-        add_custom_target(${targetName} ALL DEPENDS ${targetName}-stamp)
-    endforeach()
-endmacro()
-
-# build
-set(v1.0Definitions
-    ASLUAV.xml
-    ardupilotmega.xml
-    common.xml
-    matrixpilot.xml
-    minimal.xml
-    paparazzi.xml
-    python_array_test.xml
-    test.xml
-    ualberta.xml
-    )
-generateMavlink("1.0" "${v1.0Definitions}")
-generateMavlink("2.0" "${v1.0Definitions}")
-
-# testing
-if (BUILD_TEST)
-    if (UNIX) 
-        include_directories(${CMAKE_BINARY_DIR}/include/v1.0/common)
-        # TODO fix udp example
-        #add_executable(mavlink_udp examples/linux/mavlink_udp.c)
-    endif()
+if (NOT MAVLINK_DIALECT)
+    set(MAVLINK_DIALECT common)
 endif()
+message(STATUS "MAVLink dialect: ${MAVLINK_DIALECT}")
 
-# install files
-install(DIRECTORY ${CMAKE_BINARY_DIR}/include/ DESTINATION include/${PROJECT_NAME} COMPONENT Dev FILES_MATCHING PATTERN "*.h*")
-install(DIRECTORY ${CMAKE_BINARY_DIR}/src/ DESTINATION share/${PROJECT_NAME} COMPONENT Dev FILES_MATCHING PATTERN "*.c*")
-install(DIRECTORY ${MAVLINK_SOURCE_DIR}/share/${PROJECT_NAME} DESTINATION share COMPONENT Dev FILES_MATCHING PATTERN "*.c*")
-if (INSTALL_PYMAVLINK)
-    if (UNIX)
-        install(DIRECTORY ${MAVLINK_SOURCE_DIR}/pymavlink DESTINATION ${PYTHON_SITELIB} COMPONENT Dev)
-    else ()
-        install(DIRECTORY ${MAVLINK_SOURCE_DIR}/pymavlink DESTINATION "share/pyshared" COMPONENT Dev)
-    endif ()
+if (NOT MAVLINK_VERSION)
+    set(MAVLINK_VERSION 2.0)
 endif()
+message(STATUS "MAVLink version: ${MAVLINK_VERSION}")
+
+set(EXAMPLE_HEADER ${CMAKE_CURRENT_BINARY_DIR}/include/mavlink/${MAVLINK_DIALECT}/mavlink.h)
+
+add_custom_command(OUTPUT ${EXAMPLE_HEADER}
+    COMMAND Python::Interpreter
+        -m pymavlink.tools.mavgen
+        --lang=C
+        --wire-protocol=${MAVLINK_VERSION}
+        --output ${CMAKE_CURRENT_BINARY_DIR}/include/mavlink/
```

@julianoes: I am just realizing now the extent of changing this: it will break many includes in MAVSDK and in the code using mavlink-passthrough. I guess essentially all the code already using MAVLink a bit everywhere.

Would this work instead? I have kind of forgotten why we didn't do it in the first place, though I think I remember that you told me :see_no_evil:.

```suggestion
        --output ${CMAKE_CURRENT_BINARY_DIR}/include/mavlink/v2.0
```

---

Hm, I thought it only requires the change here: https://github.com/mavlink/MAVSDK/blob/main/src/mavsdk/core/include/mavsdk/mavlink_include.h.in#L3

---

Oh, you're right, I had missed that :+1:. Sorry :see_no_evil: